networkx/generators/tests/test_random_graphs.py

```diff
@@ -476,3 +476,34 @@ def test_random_kernel_disallow_directed_and_multigraph(graphtype):
         nx.random_kernel_graph(
             10, lambda y, a, b: a + b, lambda u, w, r: r + w, create_using=graphtype
         )
+
+
+# Test structure and connectivity for multiple (d, n, k) combinations
+@pytest.mark.parametrize(
+    "d, n, k",
+    [
+        (4, 8, 4),  # Balanced
+        (1, 2, 1),  # Tiny graph
+        (6, 20, 3),  # Higher n with moderate d
+        (80, 100, 10),  # Large-scale performance & correctness
+    ],
+)
+def test_random_k_lift_size_and_structure(d, n, k):
+    G = nx.random_regular_graph(d, n, seed=42)
+    H = nx.random_k_lift(G, k)
```

Comment:
Let's add a seed here for reproducibility.
```suggestion
    H = nx.random_k_lift(G, k, seed=42)
```